From: =?utf-8?q?Jeremy_B=C3=ADcha?= <jbicha@ubuntu.com>
Date: Sun, 9 Aug 2026 14:19:51 +0200
Subject: Revert "users: Disable keeping data if deleting homed user"

Revert "users: Disable auto-login for homed users"

This reverts commit b54ee983acba3180f346f33084f50e2ee0a95c02.
This reverts commit abdfb2ca3ace1dac047f898de3cc872eed9602f7.

Debian Unstable does not yet have accountsservice 26
---
 meson.build                          |  3 +--
 panels/system/users/cc-user-page.blp |  1 +
 panels/system/users/cc-user-page.c   | 18 +-----------------
 3 files changed, 3 insertions(+), 19 deletions(-)

diff --git a/meson.build b/meson.build
index ef4ea76..9777d36 100644
--- a/meson.build
+++ b/meson.build
@@ -222,8 +222,7 @@ gmobile_dep = dependency(
 config_h.set('HAVE_GMOBILE', gmobile_dep.found(),
              description: 'Define to 1 if gmobile is available')
 
-accounts_dep = dependency('accountsservice', version: '>= 26.27.3')
-
+accounts_dep = dependency('accountsservice', version: '>= 23.11.69')
 colord_dep = dependency('colord', version: '>= 0.1.34')
 gdk_pixbuf_dep = dependency('gdk-pixbuf-2.0', version: '>= 2.23.0')
 gio_dep = dependency('gio-2.0')
diff --git a/panels/system/users/cc-user-page.blp b/panels/system/users/cc-user-page.blp
index ecbd13c..554500a 100644
--- a/panels/system/users/cc-user-page.blp
+++ b/panels/system/users/cc-user-page.blp
@@ -188,6 +188,7 @@ template $CcUserPage: Adw.NavigationPage {
 }
 
 Adw.AlertDialog remove_local_user_dialog {
+  body: _("The user will not be able to use this device once their account has been removed");
   default-response: "cancel";
   close-response: "cancel";
   response::remove => $remove_local_user_response(template);
diff --git a/panels/system/users/cc-user-page.c b/panels/system/users/cc-user-page.c
index caa4c6c..6ce3bc7 100644
--- a/panels/system/users/cc-user-page.c
+++ b/panels/system/users/cc-user-page.c
@@ -148,15 +148,13 @@ would_demote_only_admin (ActUser *user)
 static gboolean
 get_autologin_possible (ActUser *user)
 {
-    gboolean homed;
     gboolean locked;
     gboolean set_password_at_login;
 
-    homed = act_user_uses_homed (user);
     locked = act_user_get_locked (user);
     set_password_at_login = (act_user_get_password_mode (user) == ACT_USER_PASSWORD_MODE_SET_AT_LOGIN);
 
-    return !(homed || locked || set_password_at_login);
+    return !(locked || set_password_at_login);
 }
 
 static gchar *
@@ -424,23 +422,9 @@ remove_local_user_response (CcUserPage *self)
 static void
 remove_user (CcUserPage *self)
 {
-    gboolean homed = act_user_uses_homed (self->user);
-
     // TODO: Handle enterprise accounts
     adw_alert_dialog_format_heading (self->remove_local_user_dialog, _("Remove %s?"),
                                                                        get_real_or_user_name (self->user));
-
-    if (homed)
-        adw_alert_dialog_set_body (self->remove_local_user_dialog,
-                                   _("The user's files and settings will be deleted, and they will not be able to use this device once their account has been removed"));
-    else
-        adw_alert_dialog_set_body (
-            self->remove_local_user_dialog,
-            _("The user will not be able to use this device once their account has been removed"));
-
-    adw_switch_row_set_active (self->remove_local_files_choice, homed);
-    gtk_widget_set_visible (adw_alert_dialog_get_extra_child (self->remove_local_user_dialog), !homed);
-
     adw_dialog_present (ADW_DIALOG (self->remove_local_user_dialog), GTK_WIDGET (self));
 }
 
